### PR TITLE
Supports version 2 lockfiles

### PR DESCRIPTION
**Pull request by dburles**
This PR adds support for npm 7+ as `npm-check-engines` would crash on projects with a version 2 lockfile. The programmatic npm API was dropped in version 8, so I've used a child process instead. This is a breaking change with a minimum node version of 12.13.0 and minimum npm version 7.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,11 +19,11 @@
   "homepage": "https://github.com/jgillich/npm-check-engines#readme",
   "dependencies": {
     "colors": "^1.4.0",
-    "npm": "^6.14.14",
+    "npm": "^8.1.3",
```
Comment (jaydenseric) [inline package.json]: Should `npm` be removed from `dependencies`? It's not imported or required. Dependencies are installed every time, and I'm not sure if a local installation would even be the npm CLI that gets used.

File: index.js [hunk 2]
```diff
@@ -3,27 +3,23 @@
...
+    handleErr(new Error('engines.node is not defined'));
+  }
+  engines(root.engines.node, function (err, engines) {
     handleErr(err);
-    if (!root.engines || !root.engines.node) {
-      handleErr(new Error('engines.node is not defined'));
+    if (!check(engines, root.dependencies)) {
+      console.log('no problems detected'.green);
+    } else {
+      process.exit(1);
```
Comment (jaydenseric) [inline index.js]: I know this is the way it was before the PR, but here is a more correct approach:

```suggestion
      process.exitCode = 1;
```

It's an anti-pattern to force an immediate exit of the process.

File: index.js [hunk 3]
```diff
@@ -3,27 +3,23 @@
...
 
-npm.load(config, function (err, npm) {
+execFile('npm', ['ls', '--json', '--all', '--long'], function (err, stdout, stderr) {
   handleErr(err);
-  npm.commands.ls([], true, function (err, root, lite) {
+  const root = JSON.parse(stdout);
+  if (!root.engines || !root.engines.node) {
+    handleErr(new Error('engines.node is not defined'));
+  }
+  engines(root.engines.node, function (err, engines) {
```
Comment (jaydenseric) [inline index.js:16]:
```suggestion
  engines(root.engines.node, (err, engines) => {
```

**Review by jaydenseric — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (dburles) [reply package.json]: Good point.

Comment (jaydenseric) [reply package.json]: The correct thing to do is to add `npm` to the package `engines` field, reflecting what versions support how the `npm` CLI is being used by this package.

https://github.com/jgillich/npm-check-engines/blob/3488f63234bc606c5a3a1d28f3867b3765c9dcd6/package.json#L26-L28

Comment (dburles) [reply package.json]: Yeah I was literally just looking at doing that.

File: index.js (hunk 3 above)
Comment (dburles) [reply index.js:16]: I think I'll leave this as-is for consistency

**Comment by dburles** [comment]
Ping @jgillich! This is an under appreciated utility that we're definitely still using!

**Comment by jgillich** [comment]
LGTM, sorry for the delay

**Pull request merged**